### PR TITLE
[MM-18868] Server-side filtering of epic link searching

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

The user can now search for epics server-side, limited to 50 items in the response. The search allows for the user to enter the exact issue key for the epic.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-18868

### CODE REVIEW

File: server/issue.go [hunk 1]
```diff
@@ -453,20 +480,15 @@ func httpAPIGetSearchIssues(ji Instance, w http.ResponseWriter, r *http.Request)
 
 	wg.Wait()
 
-	// We only need to send down a summary of the data
-	type issueSummary struct {
-		Value string `json:"value"`
-		Label string `json:"label"`
-	}
-	var result []issueSummary
+	var result []ReactSelectOption
```
Comment (levb) [inline server/issue.go]: 👍

**Comment by levb** [comment]
@mickmister 1/5 the server-side implementations of issue search look REALLY similar to me. Anyway, that'd be a separate ticket.

**Comment by mickmister** [comment]
@levb I have the duplication fixed, but I'll wait until after the release so we don't affect the attach comment modal.

**Comment by DHaussermann** [comment]
Searching does not support spaces in text. For ex: `Pickles in a jar` after the 1st space no results are available.

As per @mickmister there is a way to solve this issue. He will investigate.

**Pull request merged**